### PR TITLE
Move community Github and Board widgets from devgogigs to devhub

### PR DESCRIPTION
**Pull request by Megha-Dev-19**
Preview [link](https://near.org/megha19.near/widget/app?page=community&handle=devhub-test&tab=Github)
_Resolves #364_
_Resolves #575_
_Resolves #539_
_Resolves #582_

- Migrated components, refactored the code, made few updates to the UI

### CODE REVIEW

**Comment by Megha-Dev-19** [comment]
@Tguntenaar @ori-near @ailisp can you please once test out the updated UI and let me know your feedback.
Accordingly I will update kanban board as well.

**Comment by Tguntenaar** [comment]
@Megha-Dev-19 the UI looks good. However, there are some errors in the console that result in a flash of errors upon loading

File: src/core/adapter/devhub-contract.jsx
Comment (Tguntenaar) [inline src/core/adapter/devhub-contract.jsx:66]: `update_community_board` can also be removed since we now use `update_addon`.

File: src/devhub/page/addon.jsx
Comment (Tguntenaar) [inline src/devhub/page/addon.jsx:96]: Is this meant to be commented?

File: src/devhub/page/addon.jsx
Comment (Tguntenaar) [inline src/devhub/page/addon.jsx:105]: this does not work you need to save the result from the first replace in a variable

**Review by Tguntenaar — CHANGES_REQUESTED** [review]
Overall, it looks good. I just shared a few minor comments.

File: src/core/adapter/devhub-contract.jsx
Comment (Megha-Dev-19) [reply src/core/adapter/devhub-contract.jsx:66]: yeah I will remove the unused code after updating kanban board as well

File: src/core/lib/url.jsx
Comment (Megha-Dev-19) [reply src/core/lib/url.jsx:43]: oh I see, I will check if we can remove it

File: src/devhub/page/addon.jsx
Comment (Megha-Dev-19) [reply src/devhub/page/addon.jsx:105]: yeah I did that, didn't push the change yet
I will remove this after the changes are complete, will add a comment to change the contract data itself

File: src/core/lib/url.jsx
Comment (Megha-Dev-19) [reply src/core/lib/url.jsx]: I have removed it

**Comment by Megha-Dev-19** [comment]
@ailisp @Tguntenaar can we update contract data for the following: (once this PR is approved)
- move from `devgovgigs` to `devhub.near` widget for `github` (`Viewer, Configurator, kanban_ticket, kanban_board`)
- move from `devgovgigs` to `devhub.near` widget for `kanban` (`Viewer, Configurator, post_ticket, post_board`)
- update `kanban_ticket` to `KanbanTicket`, similarly for `kanban_board, post_ticket, post_board.`

**Comment by ailisp** [comment]
> @ailisp @Tguntenaar can we update contract data for the following: (once this PR is approved)
> 
>     * move from `devgovgigs` to `devhub.near` widget for `github` (`Viewer, Configurator, kanban_ticket, kanban_board`)
> 
>     * move from `devgovgigs` to `devhub.near` widget for `kanban` (`Viewer, Configurator, post_ticket, post_board`)
> 
>     * update `kanban_ticket` to `KanbanTicket`, similarly for `kanban_board, post_ticket, post_board.`

Do these need a contract upgrade? Sorry I'm a bit confused here, if you are moving widgets, I assume the CI will deploy the new widgets to devhub.near?

**Comment by Megha-Dev-19** [comment]
@ailisp the components will move to `devhub.near` but we render them according to contract data (for addons), we have the viewer, configurator, widget src defined in them, and there it is stored as `devgovgigs.near`.
so we would need to update that, to reflect the latest added component.

**Comment by ailisp** [comment]
@Megha-Dev-19 Our production contract is still deployed in devgovgigs.near. so REPL_DEVHUB is devhub.near and REPL_DEVHUB_CONTRACT is devgovgigs.near. Migrating the contract to devhub.near is not trivial and include obtain state of all posts & update on indexer, so it won't happen shortly.

**Comment by Megha-Dev-19** [comment]
hey @ailisp, I think is here is some confusion, I don't want to update all the contract data, I just want to update the addons data that too for github and kanban board only. 
For reference, Frol and Thomas were talking something similar recently [here](https://t.me/devhubplatform/5/4567).
So I just want to update `view_widget`, `configurator_widget` and `type`.
currently in code I handle it like [this](https://github.com/Megha-Dev-19/neardevhub-widgets/blob/fix-github-configurator/src/devhub/page/addon.jsx#L96), but it isn't good approach to replace it.
we can schedule a call to discuss it further if you want.

**Review by Tguntenaar — CHANGES_REQUESTED** [review]
Hey @Megha-Dev-19 ! 

Their are some merge conlficts with the main branch.
As it addresses 4 distinct issues, I have compiled a list of the acceptance criteria.

## Acceptance criteria

Based on issue #364:
- [ ] The old configurator at src/gigs-board/entity/workspace/view/github/configurator.jsx should be refactored to src/devhub/entity/addon/github/Configurator.jsx
- [ ] The configurator should work snappy

Based on issue #575:
- [x] Since [this](https://near.org/devhub.near/widget/app?page=community&handle=devhub-test&tab=Github) board is not visible it should appear in your preview [here](https://near.org/megha19.near/widget/app?page=community&handle=devhub-test&tab=Github)

Based on issue #539:

- [ ] No more error on the board on page load

Based on issue #582:

- [x] kanban addon preview should work, when "approved amount" is enabled
- [x] kanban board can be saved, it should not hit contract method not found error. Update smart contract if needed, or call an existing method if contract can already support this.

## Feedback

#364 If I edit the kanban board with more than 3 columns the state update will have a significant delay. Issue 364 was made to remove this behaviour and make it work snappy so I don't think this issue is fully resolved.

#575 is resolved

#539 While the `String.prototype.replace` error no longer appears during page load, there is still a flash of another error indicated in the console: `.map is not a function`. This occurs on the [specified page](https://near.org/megha19.near/widget/app?page=community&handle=devhub-test&tab=Github).

#582 is resolved

Based on this review we could go in two directions. Either we unlink the issues that haven't been resolved and merge the two resolved issues or we fix the other issues before we merge, your call.

**Comment by Megha-Dev-19** [comment]
Hey @Tguntenaar 
Thanks for the thorough review.
I think the first point is addressed, we have moved the widget here, we just need to update the contract data (if that’s what you mean)

For delay, could you help me know what else can I update.
I already refactored the way we update state and fetch data and separated viewer and editor. I can’t think of anything else.

also for error in console, if you have some time, can you try it on your local, because on local this error is not there for me, I think on deploy version it might be because of account change (from devhub to megha19)

if you have suggestions, I can work on them right away, else we can merge it and then discuss them, whatever you feel right

**Comment by Megha-Dev-19** [comment]
@Tguntenaar I have decoupled the state, it has improved the performance pretty well.

File: src/core/lib/url.jsx
Comment (Tguntenaar) [reply src/core/lib/url.jsx]: What about this part? If it's not used we can rid of it as well

**Review by Tguntenaar — CHANGES_REQUESTED** [review]
It looks good, great job! Just one more error on page load which states: normalizeLabel is not a function. The 3 comments I made before on the unused code, the github comment and the function `updateWidgetEndpoint` are not processed yet I think.

File: src/core/lib/url.jsx
Comment (Megha-Dev-19) [reply src/core/lib/url.jsx]: it's removed, `widget` is not there anymore

**Comment by Megha-Dev-19** [comment]
@elliotBraem @Tguntenaar I have scheduled a call tomorrow with @ailisp, so we will update the contract data and UI then, can you please review the PR and approve it if it all looks good?

**Review by Tguntenaar — APPROVED** [review]
Yes all looks good!

**Comment by Megha-Dev-19** [comment]
hey @Tguntenaar @elliotBraem, we can merge it, the contract data is updated.

**Pull request merged**
